developer-docs-site/static/move_demo/sources/ACLMessageBoard.move

```diff
@@ -0,0 +1,98 @@
+
+
+    struct Messageboard has key, store {
+        participants: ACL::ACL,
+        notice: vector<u8>
+    }
+
+    struct NoticeChangeEventHandle has key, store {
+        change_events: EventHandle<NoticeChangeEvent>
+    }
+    struct NoticeChangeEvent has store, drop {
+        timestamp: u64,
+        sender: address,
+        message: vector<u8>,
+        board: address
+    }
+
+    /// init the message board and move the resource to signer
+    public fun message_board_init_internal(account: &signer){
+        let board = Messageboard {
+            participants: ACL::empty(),
+            notice: Vector::empty<u8>()
+        };
+        ACL::add(&mut board.participants, Signer::address_of(account));
+        move_to(account, board);
+    }
+
+    /// expose the function as a transaction interface
+    public(script) fun message_board_init(account: signer) {
+        message_board_init_internal(&account);
+    }
+
+    public fun view_message(board_addr: address) : vector<u8> acquires Messageboard {
+        let board = borrow_global<Messageboard>(board_addr);
+        board.notice
+    }
+
+    /// participant can register to
+    public fun register_internal(participant: &signer){
+        move_to(participant, NoticeChangeEventHandle {
+            change_events:  Event::new_event_handle<NoticeChangeEvent>(participant)
+        })
+    }
+
+    public(script) fun register(participant: signer) {
+        register_internal(&participant);
+    }
+
+    /// board owner control adding new participants
+    public fun add_participant_internal(account: &signer, participant: address) acquires Messageboard {
+        let board = borrow_global_mut<Messageboard>(Signer::address_of(account));
+        ACL::add(&mut board.participants, participant);
+    }
+
+    public(script) fun add_participant(account: signer, participant: address) acquires Messageboard {
+        add_participant_internal(&account, participant);
+    }
+
+    /// remove a participant from the ACL
+    public(script) fun remove_participant(account: signer, participant: address) acquires Messageboard {
+        let board = borrow_global_mut<Messageboard>(Signer::address_of(&account));
+        assert!(Signer::address_of(&account) != participant, CANNOT_REMOVE_ADMIN_FROM_ACL);
+        ACL::remove( &mut board.participants, participant);
+    }
+
+    /// an account publish the message to the update the notice
+    public fun submit_pinned_message_internal(account: &signer,  board_addr: address, message: vector<u8>) acquires Messageboard, NoticeChangeEventHandle {
+
+        let board = borrow_global<Messageboard>(board_addr);
+        assert!(ACL::contains(&board.participants, Signer::address_of(account)), ACCOUNT_NOT_IN_ACL);
+        let board = borrow_global_mut<Messageboard>(board_addr);
+        let send_acct = Signer::address_of(account);
+        let event_handle = borrow_global_mut<NoticeChangeEventHandle>(send_acct);
+        board.notice = message;
```

Comment:
probably reorder and add whitespace

we get the board on 81, let's set the message on line 82, let's move the event logic to 83.